src/compiler/checker.ts

```diff
@@ -6192,15 +6195,27 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return context.encounteredError ? undefined : resultingNode;
         }
 
+        function saveFlags(context: NodeBuilderContext) {
+            const flags = context.flags;
+            const internalFlags = context.internalFlags;
+
+            return { restore };
+
+            function restore() {

```

Comment:
This is a good idea, but could probably just return the restore function itself to avoid the object.

Reply:
Done. I changed the name to saveRestoreFlags as it makes more sense to me.